presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -419,8 +420,17 @@ private static void verifyFileHasColumnNames(List<OrcColumn> columns, Path path)
     static void verifyAcidSchema(OrcReader orcReader, Path path)
     {
         OrcColumn rootColumn = orcReader.getRootColumn();
-        if (rootColumn.getNestedColumns().size() != 6) {
-            throw new PrestoException(HIVE_BAD_DATA, "ORC ACID file should have 6 columns: " + path);
+        List<OrcColumn> nestedColumns = rootColumn.getNestedColumns();
+        if (nestedColumns.size() != 6) {
+            throw new PrestoException(
+                    HIVE_BAD_DATA,
+                    format(
+                            "ORC ACID file should have 6 columns, found %s %s in %s",
+                            nestedColumns.size(),
+                            nestedColumns.stream()
+                                    .map(column -> format("%s (%s)", column.getColumnName(), column.getColumnType()))
```

Comment:
The only concern I have about this one is if we cannot maybe leak some information, we should not, to user by making this message such verbose.

Reply:
Why not to log on debug additional information?

Reply:
Because it will make the additional information hard to access.